+ CSES Problems (#619)
cpinitiative/usaco-guide: solutions/CSES Increasing Array Queries.mdx

---
id: cses-2416
source: CSES
title: Increasing Array Queries
author: Andi Qu
---

**Time Complexity:** $\mathcal O((N + Q) \log N)$.

> Since this is CSES, we can answer the queries offline (i.e. read in the queries and process them in a different order). More specifically, we process queries in order of their left endpoints.

Author: ```suggestion
In this problem, we can answer the queries offline (i.e. read in the queries and process them in a different order). More specifically, we process queries in order of their left endpoints.
```
idk if you should use this suggestion
